src/main/java/liquibase/ext/spanner/sqlgenerator/CreateDatabaseChangeLogLockTableGeneratorSpanner.java

```diff
@@ -34,14 +36,40 @@ public class CreateDatabaseChangeLogLockTableGeneratorSpanner
           + "    lockedby    string(max),\n"
           + ") primary key (id)";
 
+  final String createPostgresqlTableSQL =
+      ""
+          + "CREATE TABLE public.__DATABASECHANGELOGLOCK__\n"
+          + "(\n"
+          + "    id          bigint primary key,\n"
+          + "    locked      bool,\n"
+          + "    lockgranted timestamptz,\n"
+          + "    lockedby    varchar(255)\n"
+          + ")";
+
   @Override
   public Sql[] generateSql(
       CreateDatabaseChangeLogLockTableStatement statement,
       Database database,
       SqlGeneratorChain sqlGeneratorChain) {
+
+    Dialect dialect = null;
+    try {
+      dialect = ((ICloudSpanner) database).getDialect();
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
     String databaseChangeLogLockTableName = getDatabaseChangeLogLockTableNameFrom(database);
-    String createTableSQL =
-        this.createTableSQL.replaceAll("__DATABASECHANGELOGLOCK__", databaseChangeLogLockTableName);
+    String createTableSQL = "";
+    if (dialect == Dialect.GOOGLE_STANDARD_SQL) {
+      createTableSQL =
+          this.createTableSQL.replaceAll(
+              "__DATABASECHANGELOGLOCK__", databaseChangeLogLockTableName);
+    } else if (dialect == Dialect.POSTGRESQL) {
+      createTableSQL =
+          this.createPostgresqlTableSQL.replaceAll(
+              "__DATABASECHANGELOGLOCK__", databaseChangeLogLockTableName);
+    }
```

Comment:
nit, something like this would be preferred (and a bit shorter):

```suggestion
    String createTableSQL = dialect == Dialect.POSTGRESQL
        ? this.createPostgresqlTableSQL
        : this.createTableSQL;
    createTableSQL = createTableSQL.replaceAll(
        "__DATABASECHANGELOGLOCK__",
        databaseChangeLogLockTableName);
```

The reason that this would be preferred is that we should treat `POSTGRESQL` as the 'exception' and GoogleSQL as the default. This ensures that if we were to introduce a new dialect in the future, then this would ensure that this code would still produce at least partially usable SQL. So in general:
1. Use PostgreSQL dialect when the dialect == POSTGRESQL.
2. Use GoogleSQL in all other cases.